cpp/src/copying/scatter.cu

```diff
@@ -329,6 +332,104 @@ std::vector<std::unique_ptr<table>> scatter_to_tables(
+    CUDF_EXPECTS(boolean_mask.size() == target.num_rows(), "Boolean mask size and number of target rows mismatch");
+    CUDF_EXPECTS(boolean_mask.type().id() == BOOL8, "Mask must be of Boolean type");
+    std::vector<cudf::size_type> column_indices(target.num_columns());
+    std::iota(std::begin(column_indices), std::end(column_indices), 0);
+    // Count valid pair of source and columns as per type at each column index i
+    auto valid_pair_count = std::count_if(column_indices.begin(), column_indices.end(), [&source, &target](auto index){
+                return ((source.column(index).type().id()) == (target.column(index).type().id()));
+                });
+   CUDF_EXPECTS(target.num_columns() == valid_pair_count,"Type mismatch in source scalar and target column");
+
+    if (target.num_rows() != 0) {
+        std::vector<std::unique_ptr<column>> out_columns(target.num_columns());
+        std::transform(source.begin(), source.end(), target.begin(),
+                out_columns.begin(),
+                [&boolean_mask, mr, stream] (auto const& source_column, auto const& target_column){
+                return boolean_mask_scatter(source_column, target_column,
+                        boolean_mask, mr, stream);});
+
+        return std::make_unique<experimental::table>(std::move(out_columns));
+    } else {
+        return experimental::empty_like(target);
+    }
+}
+
+std::unique_ptr<table> boolean_mask_scatter(
+    std::vector<std::unique_ptr<scalar>> const& source,
+    table_view const& target,
+    column_view const& boolean_mask,
+    rmm::mr::device_memory_resource *mr,
+    cudaStream_t stream) {
+
+    CUDF_EXPECTS(static_cast<size_type>(source.size()) == target.num_columns(), "Mismatch in number of scalars and target columns");
+    CUDF_EXPECTS(boolean_mask.size() == target.num_rows(), "Boolean mask size and number of target rows mismatch");
+    CUDF_EXPECTS(boolean_mask.type().id() == BOOL8, "Mask must be of Boolean type");
+
+    std::vector<cudf::size_type> column_indices(target.num_columns());
+    std::iota(std::begin(column_indices), std::end(column_indices), 0);
+    // Count valid pair of source and columns as per type at each column/scalar index i
+    auto valid_pair_count = std::count_if(column_indices.begin(), column_indices.end(), [&source, &target](auto index){
+                return ((source[index]->type().id()) == (target.column(index).type().id()));
+                });
+   CUDF_EXPECTS(target.num_columns() == valid_pair_count,"Type mismatch in source scalar and target column");
+
+    if (target.num_rows() != 0) {
+        std::vector<std::unique_ptr<column>> out_columns(target.num_columns());
+        std::transform(source.begin(), source.end(), target.begin(),
+                out_columns.begin(),
+                [&boolean_mask, mr, stream] (auto const& scalar, auto const& target_column){
+                return boolean_mask_scatter(*(scalar.get()), target_column,
```

Comment:
Why do you have to .get()? Can you just pass or forward the scalar?

Reply:
scalar here is a unique pointer